ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+/**
+ * @return resource|false
+ */
+function imap_open(string $mailbox, string $user, string $password, int $options = 0, int $n_retries = 0) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_reopen($stream_id, string $mailbox, int $options = 0, int $n_retries = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_close($stream_id, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_num_msg($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return int|false
+ */
+function imap_num_recent($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_headers($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_headerinfo($stream_id, int $msg_no, int $from_length = 0, int $subject_length = 0, string $default_host = '') {}
+
+function imap_rfc822_parse_headers(string $headers, string $default_host = ''): object {}
```

Comment:
Uh, funny, that should be
```suggestion
function imap_rfc822_parse_headers(string $headers, string $default_host = 'UNKNOWN'): object {}
```
Note that UNKNOWN is a string here.